third_party/blink/renderer/core/record_replay/lifecycle.cc

```diff
@@ -0,0 +1,135 @@
+
+#include "base/record_replay_paint_surface.h"
+#include "third_party/blink/renderer/core/frame/local_frame.h"
+#include "third_party/blink/renderer/core/page/focus_controller.h"
+#include "third_party/blink/renderer/core/page/page.h"
+#include "third_party/blink/renderer/core/record_replay/lifecycle.h"
+#include "third_party/blink/renderer/platform/wtf/casting.h"
+
+#define LOG_LIFECYCLE_EVENTS 1
+
+namespace recordreplay {
+
+// the current set of ordinary pages, presumably scoped to this renderer process
+// (TODO verify this last part) lives at blink::Page::OrdinaryPages().  We
+// should be able to use this to get the JS contexts for each main frame, but I
+// haven't traced through.
+
+// There might be a reference to our last-active page someplace in blink
+// already, but until/unless we find it, we'll update it here.  It should be
+// safe to keep the raw pointer to the page, since we'll clear it when the page
+// is destroyed.
+static blink::Page* last_active_page = nullptr;
+
+static void set_last_active_page(blink::Page* page) {
```

Comment:
I don't think we nor Chromium uses `snake_case` for functions anywhere. Can we change this to `PascalCase`?